Dockerfile

```diff
@@ -10,7 +10,7 @@ RUN export DEBCONF_NONINTERACTIVE_SEEN=true \
     apt-get -yyqq install ca-certificates ; \
     apt-get clean
 
-FROM --platform=$TARGETPLATFORM scratch
+FROM scratch
 
 COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```

Comment:
This dockerfile is essentially re-creating distroless without tzdata and then adding coredns..

If we started FROM distroless and only ran ADD corens / EXPOSE / ENTRYPOINT, you won't have any emulated build steps, but can still use buildkit

https://github.com/GoogleContainerTools/distroless/tree/main/base#documentation-for-gcriodistrolessbase-and-gcriodistrolessstatic

> Statically compiled applications (Go) that do not require libc can use the gcr.io/distroless/static image, which contains:
> ca-certificates
> A /etc/passwd entry for a root user
> A /tmp directory
> tzdata

Even if we didn't want things other than cacerts, doing this should also avoid emulation:
```dockerfile
FROM gcr.io/distroless/static
FROM scratch
COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
ADD coredns /coredns
EXPOSE 53 53/udp
ENTRYPOINT ["/coredns"]
```
Since we just need to avoid `RUN` (which is also harder to reproduce), and shouldn't really need to just to get an image with cacerts.

Reply:
It seems that `distroless` does not have `linux/mips64le` support. A little bit of modification on the first one should get the job done without error (though I doubt if anyone actually uses `linux/mips64le`):
```
FROM --platform=$BUILDPLATFORM gcr.io/distroless/static
FROM --platform=$TARGETPLATFORM scratch
COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
ADD coredns /coredns
EXPOSE 53 53/udp
ENTRYPOINT ["/coredns"]
```

Reply:
Good point!

If someone is also _building_ on linux/mips64le (I also have doubts) we can fix with explicitly picking a fixed platform we know to be supported to grab the certs from:
```dockerfile
FROM --platform=linux/amd64 gcr.io/distroless/static
FROM --platform=$TARGETPLATFORM scratch
COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
ADD coredns /coredns
EXPOSE 53 53/udp
ENTRYPOINT ["/coredns"]
```